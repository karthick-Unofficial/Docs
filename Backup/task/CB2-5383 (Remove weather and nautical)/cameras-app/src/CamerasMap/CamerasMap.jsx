import React, { useCallback, useEffect, useState, memo } from "react";

// Mapbox
import ReactMapboxGl, { RotationControl, ZoomControl } from "react-mapbox-gl";

// Components
import CamerasMapLayersContainer from "./CamerasMapLayers/CamerasMapLayersContainer";
// Map Styles
import aresTileServer from "./mapstyles/ares/aresTileServer.json";

import { ContextMenuContainer } from "orion-components/Map/ContextMenu";
import ErrorBoundary from "orion-components/ErrorBoundary";

// Mapbox doesn't support relative urls for sprite paths, so we have to grab the correct hostname here and set manually
import { setRTLTextPlugin } from "mapbox-gl";

const setSpritePath = path =>
	path.replace(
		"please-replace-me-with-the-correct-hostname",
		location.hostname
	);

aresTileServer.sprite = setSpritePath(aresTileServer.sprite);

const Map = ReactMapboxGl({
	accessToken:
		"pk.eyJ1IjoibGNsYXJrc29uIiwiYSI6ImNpaWh2dzlnaDAwZmp1ZGtueXdpa2loNXcifQ.gcbR_GT1AD2kUmIxkSxseA"
});

const MapBaseMapboxGL = ({
	setMapReference,
	zoom,
	center,
	WavCamOpen,
	baseMaps,
	style,
	setMapPosition,
	mapRef }) => {

	useEffect(() => {
		setRTLTextPlugin(
			"https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-rtl-text/v0.2.3/mapbox-gl-rtl-text.js",
			null,
			true
		);
	}, []);
	const [mapState, setMapState] = useState(null);


	const getStyle = useCallback(() => {
		if (baseMaps.length > 0) {
			const baseMapSelection = baseMaps.filter((element) => element.name === style)[0];
			const spritePath = baseMapSelection.style;
			spritePath.sprite = setSpritePath(spritePath.sprite);
			return spritePath;
		}

	}, [baseMaps, style]);

	const setMap = map => {
		// Only on first style load
		setMapReference(map);
		// Set up future style loads to reinsert layers
		// map.on("style.load", () => {
		// 	// enableLayers();
		// });
		setMapState(map);
	};

	const setPosition = map => {
		// convert center object to array
		const newCoords = Object.values(map.getCenter());
		setMapPosition(newCoords, map.getZoom());
	};

	function getToken() {
		return "pk.eyJ1IjoibGNsYXJrc29uIiwiYSI6ImNpaWh2dzlnaDAwZmp1ZGtueXdpa2loNXcifQ.gcbR_GT1AD2kUmIxkSxseA";
	}

	return (
		<Map
			/* Comment out accessToken and style below to use internal tile server */
			style={getStyle()}
			center={center}
			maxZoom={18}
			minZoom={2}
			zoom={[zoom]}
			/* Uncomment this line to use local settings for tile server */
			/* style={simple} */
			containerStyle={{ width: "100vw", height: `calc(100vh - ${WavCamOpen ? "288px" : "48px"})` }}
			onStyleLoad={map => setMap(map)}
			className="map-base"
			movingMethod="easeTo"
			onMoveEnd={map => setPosition(map)}
		>
			<ZoomControl />
			<RotationControl />
			{mapState && (
				<ErrorBoundary>
					<CamerasMapLayersContainer map={mapState} />
					<ContextMenuContainer map={mapState} />
				</ErrorBoundary>
			)}
		</Map>
	);
};

const shouldComponentUpdate = (prevProps, nextProps) => {
	if (prevProps.mapRef && prevProps.WavCamOpen === nextProps.WavCamOpen) {
		return true;
	} else {
		return false;
	}
};

export default memo(MapBaseMapboxGL, shouldComponentUpdate);