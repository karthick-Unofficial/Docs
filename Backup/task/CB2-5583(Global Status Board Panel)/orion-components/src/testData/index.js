export { activityData, activityDataArr } from "./activityData";
export { applicationData, applicationDataArr } from "./applicationData";
export { attachmentData, attachmentDataArr } from "./attachmentData";
export { entityCollectionData, entityCollectionDataArr } from "./entityCollectionData";
export { entityStoreData, entityStoreDataArr } from "./entityStoreData";
export { feedEntitiesData, feedEntitiesDataArr } from "./feedEntitiesData";
export { notificationData, notificationDataArr } from "./notificationData";
export { organizationData, organizationDataArr } from "./organizationData";
export { orgRolesData, orgRolesDataArr } from "./orgRolesData";
export { userDeviceData, userDeviceDataArr } from "./userDeviceData";
export { userData, userDataArr } from "./userData";
export { userProfileData } from "./userProfileData";

