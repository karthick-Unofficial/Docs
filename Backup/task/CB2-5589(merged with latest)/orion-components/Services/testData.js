(function (global, factory) {
	if (typeof define === "function" && define.amd) {
		define(["exports"], factory);
	} else if (typeof exports !== "undefined") {
		factory(exports);
	} else {
		var mod = {
			exports: {}
		};
		factory(mod.exports);
		global.testData = mod.exports;
	}
})(this, function (exports) {
	"use strict";

	Object.defineProperty(exports, "__esModule", {
		value: true
	});
	exports.default = {
		"core-system": {
			"label": "Core System",
			"type": "core-system",
			"systems": [{
				"label": "Ecosystem/Core",
				"alwaysShow": false,
				"checks": [{
					"id": "ecosystem-app-instance-availability",
					"name": "Ecosystem Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.651Z"
				}, {
					"id": "external-system-processor-process-instance-availability",
					"name": "External System Entity Processor",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.655Z"
				}, {
					"id": "shape-activity-generator-process-instance-availability",
					"name": "Shape Activity Generator",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.653Z"
				}]
			}, {
				"label": "Databases",
				"alwaysShow": false,
				"checks": [{
					"id": "ecosystem-rethinkdb-availability",
					"name": "RethinkDB Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.668Z"
				}, {
					"id": "ecosystem-elasticsearch-availability",
					"name": "Elasticsearch Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.670Z"
				}]
			}]
		},
		"feeds": {
			"label": "My Feeds",
			"type": "feeds",
			"systems": [{
				"label": "aishub",
				"alwaysShow": true,
				"checks": [{
					"id": "ecosystem-feed-total-transactions-aishub",
					"name": "Feed aishub minimum 1 txn(s) in the last 5 minutes",
					"status": {
						"ok": true,
						"message": "Feed aishub total transaction count of 1064 in last 5 minutes is above allowed minimum of 1"
					},
					"lastUpdated": "2019-09-04T18:04:30.697Z"
				}]
			}]
		},
		"apps": {
			"label": "My Apps",
			"type": "my-apps",
			"systems": [{
				"label": "Berth Schedule App",
				"appId": "berth-schedule-app",
				"alwaysShow": false,
				"checks": [{
					"id": "berth-schedule-app-app-instance-availability",
					"name": "App Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.649Z"
				}, {
					"id": "vessel-berthing-event-processor-process-instance-availability",
					"name": "Berthing Event Processor",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.653Z"
				}, {
					"id": "berth-request-app-app-instance-availability",
					"name": "Agent Portal Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.658Z"
				}]
			}, {
				"label": "Map App",
				"appId": "map-id",
				"alwaysShow": false,
				"checks": [{
					"id": "mapgl-app-app-instance-availability",
					"name": "App Availability"
				}, {
					"id": "gis-app-app-instance-availability",
					"name": "GIS Service Availability"
				}]
			}, {
				"label": "Lists App",
				"appId": "lists-app",
				"alwaysShow": false,
				"checks": [{
					"id": "lists-app-app-instance-availability",
					"name": "App Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.642Z"
				}]
			}, {
				"label": "Events App",
				"appId": "events-app",
				"alwaysShow": false,
				"checks": [{
					"id": "events-app-app-instance-availability",
					"name": "App Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.649Z"
				}]
			}, {
				"label": "Cameras App",
				"appId": "cameras-app",
				"alwaysShow": false,
				"checks": [{
					"id": "cameras-app-app-instance-availability",
					"name": "App Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.631Z"
				}, {
					"id": "slew-lock-manager-process-instance-availability",
					"name": "Slew Lock Manager",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.633Z"
				}]
			}, {
				"label": "Settings App",
				"appId": "settings-app",
				"alwaysShow": false,
				"checks": [{
					"id": "settings-app-app-instance-availability",
					"name": "App Availability"
				}]
			}, {
				"label": "Camera Wall App",
				"appId": "camera-wall-app",
				"alwaysShow": false,
				"checks": [{
					"id": "camera-wall-app-app-instance-availability",
					"name": "App Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.636Z"
				}]
			}, {
				"label": "Law Enforcement Search App",
				"appId": "law-enforcement-search-app",
				"alwaysShow": false,
				"checks": [{
					"id": "law-enforcement-search-app-app-instance-availability",
					"name": "App Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.645Z"
				}]
			}, {
				"label": "Reports App",
				"appId": "reports-app",
				"alwaysShow": false,
				"checks": [{
					"id": "reports-app-app-instance-availability",
					"name": "App Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.639Z"
				}, {
					"id": "zone-activity-process-instance-availability",
					"name": "Zone Activity Processor",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.648Z"
				}, {
					"id": "tripwire-activity-process-instance-availability",
					"name": "Tripwire Activity Processor",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.660Z"
				}, {
					"id": "dwell-time-process-instance-availability",
					"name": "Dwell Time Processor",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.662Z"
				}, {
					"id": "feed-history-process-instance-availability",
					"name": "Feed History Processor",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.664Z"
				}]
			}, {
				"label": "Rules App",
				"appId": "rules-app",
				"alwaysShow": false,
				"checks": [{
					"id": "rules-app-app-instance-availability",
					"name": "App Availability",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.631Z"
				}, {
					"id": "rule-analyzer-process-instance-availability",
					"name": "Rule Analyzer",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.633Z"
				}, {
					"id": "loiter-activity-generator-process-instance-availability",
					"name": "Loiter Activity Generator",
					"status": {
						"ok": true
					},
					"lastUpdated": "2019-09-04T18:04:30.678Z"
				}]
			}]
		},
		"server-health": {
			"label": "Server Health",
			"type": "server-health",
			"systems": [{
				"label": "CB2 Demo1",
				"alwaysShow": true,
				"checks": [{
					"id": "ecosystem-server-cpu-CB2-Demo1",
					"name": "Server CPU",
					"status": {
						"ok": true,
						"message": "Average CPU of 13 over last 5 minutes is below CPU threshold of 50%",
						"details": {
							"host": "CB2-Demo1"
						}
					},
					"lastUpdated": "2019-09-04T18:04:30.678Z"
				}, {
					"id": "ecosystem-server-memory-CB2-Demo1",
					"name": "Server Memory",
					"status": {
						"ok": false,
						"message": "Current memory usage of 62% averaged over last 5 minutes exceed threshold of 50%",
						"details": {
							"host": "CB2-Demo1"
						}
					},
					"lastUpdated": "2019-09-04T18:04:30.682Z"
				}, {
					"id": "ecosystem-server-filesystem-CB2-Demo1",
					"name": "Server Filesystem",
					"status": {
						"ok": false,
						"message": "Disk utilization of 72% exceeds threshold of 40%",
						"details": {
							"host": "CB2-Demo1"
						}
					},
					"lastUpdated": "2019-09-04T18:04:30.686Z"
				}]
			}]
		}
	};
});