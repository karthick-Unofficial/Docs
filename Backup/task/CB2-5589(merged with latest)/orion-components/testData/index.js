"use strict";

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");
var _mapInstanceProperty = require("@babel/runtime-corejs3/core-js-stable/instance/map");
_Object$defineProperty(exports, "__esModule", {
  value: true
});
_Object$defineProperty(exports, "activityData", {
  enumerable: true,
  get: function get() {
    return _activityData.activityData;
  }
});
_Object$defineProperty(exports, "activityDataArr", {
  enumerable: true,
  get: function get() {
    return _activityData.activityDataArr;
  }
});
_Object$defineProperty(exports, "appState", {
  enumerable: true,
  get: function get() {
    return _appState.appState;
  }
});
_Object$defineProperty(exports, "applicationData", {
  enumerable: true,
  get: function get() {
    return _applicationData.applicationData;
  }
});
_Object$defineProperty(exports, "applicationDataArr", {
  enumerable: true,
  get: function get() {
    return _applicationData.applicationDataArr;
  }
});
_Object$defineProperty(exports, "assignments", {
  enumerable: true,
  get: function get() {
    return _assignments.assignments;
  }
});
_Object$defineProperty(exports, "attachmentData", {
  enumerable: true,
  get: function get() {
    return _attachmentData.attachmentData;
  }
});
_Object$defineProperty(exports, "attachmentDataArr", {
  enumerable: true,
  get: function get() {
    return _attachmentData.attachmentDataArr;
  }
});
_Object$defineProperty(exports, "berths", {
  enumerable: true,
  get: function get() {
    return _berths.berths;
  }
});
_Object$defineProperty(exports, "clientConfig", {
  enumerable: true,
  get: function get() {
    return _clientConfig.clientConfig;
  }
});
_Object$defineProperty(exports, "entityCollectionData", {
  enumerable: true,
  get: function get() {
    return _entityCollectionData.entityCollectionData;
  }
});
_Object$defineProperty(exports, "entityCollectionDataArr", {
  enumerable: true,
  get: function get() {
    return _entityCollectionData.entityCollectionDataArr;
  }
});
_Object$defineProperty(exports, "entityStoreData", {
  enumerable: true,
  get: function get() {
    return _entityStoreData.entityStoreData;
  }
});
_Object$defineProperty(exports, "entityStoreDataArr", {
  enumerable: true,
  get: function get() {
    return _entityStoreData.entityStoreDataArr;
  }
});
_Object$defineProperty(exports, "feedEntitiesData", {
  enumerable: true,
  get: function get() {
    return _feedEntitiesData.feedEntitiesData;
  }
});
_Object$defineProperty(exports, "feedEntitiesDataArr", {
  enumerable: true,
  get: function get() {
    return _feedEntitiesData.feedEntitiesDataArr;
  }
});
_Object$defineProperty(exports, "formPanel", {
  enumerable: true,
  get: function get() {
    return _formPanel.formPanel;
  }
});
_Object$defineProperty(exports, "globalData", {
  enumerable: true,
  get: function get() {
    return _globalData.globalData;
  }
});
_Object$defineProperty(exports, "globalGeo", {
  enumerable: true,
  get: function get() {
    return _globalGeo.globalGeo;
  }
});
_Object$defineProperty(exports, "i18n", {
  enumerable: true,
  get: function get() {
    return _i18n.i18n;
  }
});
_Object$defineProperty(exports, "map", {
  enumerable: true,
  get: function get() {
    return _mapInstanceProperty(_map);
  }
});
_Object$defineProperty(exports, "notificationData", {
  enumerable: true,
  get: function get() {
    return _notificationData.notificationData;
  }
});
_Object$defineProperty(exports, "notificationDataArr", {
  enumerable: true,
  get: function get() {
    return _notificationData.notificationDataArr;
  }
});
_Object$defineProperty(exports, "orgRolesData", {
  enumerable: true,
  get: function get() {
    return _orgRolesData.orgRolesData;
  }
});
_Object$defineProperty(exports, "orgRolesDataArr", {
  enumerable: true,
  get: function get() {
    return _orgRolesData.orgRolesDataArr;
  }
});
_Object$defineProperty(exports, "organizationData", {
  enumerable: true,
  get: function get() {
    return _organizationData.organizationData;
  }
});
_Object$defineProperty(exports, "organizationDataArr", {
  enumerable: true,
  get: function get() {
    return _organizationData.organizationDataArr;
  }
});
_Object$defineProperty(exports, "searchForm", {
  enumerable: true,
  get: function get() {
    return _searchForm.searchForm;
  }
});
_Object$defineProperty(exports, "session", {
  enumerable: true,
  get: function get() {
    return _session.session;
  }
});
_Object$defineProperty(exports, "userAppState", {
  enumerable: true,
  get: function get() {
    return _userAppState.userAppState;
  }
});
_Object$defineProperty(exports, "userData", {
  enumerable: true,
  get: function get() {
    return _userData.userData;
  }
});
_Object$defineProperty(exports, "userDataArr", {
  enumerable: true,
  get: function get() {
    return _userData.userDataArr;
  }
});
_Object$defineProperty(exports, "userDeviceData", {
  enumerable: true,
  get: function get() {
    return _userDeviceData.userDeviceData;
  }
});
_Object$defineProperty(exports, "userDeviceDataArr", {
  enumerable: true,
  get: function get() {
    return _userDeviceData.userDeviceDataArr;
  }
});
_Object$defineProperty(exports, "userProfileData", {
  enumerable: true,
  get: function get() {
    return _userProfileData.userProfileData;
  }
});
var _activityData = require("./activityData");
var _applicationData = require("./applicationData");
var _attachmentData = require("./attachmentData");
var _entityCollectionData = require("./entityCollectionData");
var _entityStoreData = require("./entityStoreData");
var _feedEntitiesData = require("./feedEntitiesData");
var _notificationData = require("./notificationData");
var _organizationData = require("./organizationData");
var _orgRolesData = require("./orgRolesData");
var _userDeviceData = require("./userDeviceData");
var _userData = require("./userData");
var _userProfileData = require("./userProfileData");
var _appState = require("./appState");
var _globalData = require("./globalData");
var _globalGeo = require("./globalGeo");
var _userAppState = require("./userAppState");
var _session = require("./session");
var _i18n = require("./i18n");
var _clientConfig = require("./clientConfig");
var _searchForm = require("./searchForm");
var _berths = require("./berths");
var _formPanel = require("./formPanel");
var _map = require("./map");
var _assignments = require("./assignments");