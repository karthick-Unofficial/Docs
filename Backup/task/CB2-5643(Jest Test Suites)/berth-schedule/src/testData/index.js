export { appState } from "./appState";
export { session } from "./session";
export { i18n } from "./i18n";
export { map } from "./map";
export { berths } from "./berths";
export { formPanel } from "./formPanel";
export { assignments } from "./assignments";
export { clientConfig } from "./clientConfig";