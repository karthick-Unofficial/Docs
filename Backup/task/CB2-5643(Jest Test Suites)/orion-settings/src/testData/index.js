export { appState } from "./appState";
export { session } from "./session";
export { i18n } from "./i18n";
export { globalData } from "./globalData";