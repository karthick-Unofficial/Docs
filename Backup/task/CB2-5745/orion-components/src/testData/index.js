export { activityData, activityDataArr } from "./activityData";
export { applicationData, applicationDataArr } from "./applicationData";
export { attachmentData, attachmentDataArr } from "./attachmentData";
export {
	entityCollectionData,
	entityCollectionDataArr
} from "./entityCollectionData";
export { entityStoreData, entityStoreDataArr } from "./entityStoreData";
export { feedEntitiesData, feedEntitiesDataArr } from "./feedEntitiesData";
export { notificationData, notificationDataArr } from "./notificationData";
export { organizationData, organizationDataArr } from "./organizationData";
export { orgRolesData, orgRolesDataArr } from "./orgRolesData";
export { userDeviceData, userDeviceDataArr } from "./userDeviceData";
export { userData, userDataArr } from "./userData";
export { userProfileData } from "./userProfileData";
export { appState } from "./appState";
export { globalData } from "./globalData";
export { globalGeo } from "./globalGeo";
export { userAppState } from "./userAppState";
export { session } from "./session";
export { i18n } from "./i18n";
export { clientConfig } from "./clientConfig";
export { searchForm } from "./searchForm";
export { berths } from "./berths";
export { formPanel } from "./formPanel";
export { map } from "./map";
export { assignments } from "./assignments";
