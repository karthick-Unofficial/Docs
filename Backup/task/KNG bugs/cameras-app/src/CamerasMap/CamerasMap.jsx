import React, { Component } from "react";

// Mapbox
import ReactMapboxGl, { RotationControl, ZoomControl } from "react-mapbox-gl";

// Components
import CamerasMapLayersContainer from "./CamerasMapLayers/CamerasMapLayersContainer";
// Map Styles
import aresTileServer from "./mapstyles/ares/aresTileServer.json";

import { ContextMenuContainer } from "orion-components/Map/ContextMenu";
import ErrorBoundary from "orion-components/ErrorBoundary";

// Mapbox doesn't support relative urls for sprite paths, so we have to grab the correct hostname here and set manually

const setSpritePath = path =>
	path.replace(
		"please-replace-me-with-the-correct-hostname",
		location.hostname
	);

aresTileServer.sprite = setSpritePath(aresTileServer.sprite);

const Map = ReactMapboxGl({
	accessToken:
		"pk.eyJ1IjoibGNsYXJrc29uIiwiYSI6ImNpaWh2dzlnaDAwZmp1ZGtueXdpa2loNXcifQ.gcbR_GT1AD2kUmIxkSxseA"
});

class MapBaseMapboxGL extends Component {
	constructor(props) {
		super(props);

		this.state = {
			map: null,
			mapStyle: {}
		};
	}

	static getDerivedStateFromProps(props, state) {
		if (props.baseMaps.length > 0) {
			const { style } = props;
			const baseMapSelection = props.baseMaps.filter((element) => element.name === style)[0];
			let spritePath = baseMapSelection.style;
			spritePath.sprite = setSpritePath(spritePath.sprite);
			return {
				mapStyle: spritePath
			};
		}

	}

	setMap = map => {
		// Only on first style load
		this.props.setMapReference(map);
		// Set up future style loads to reinsert layers
		// map.on("style.load", () => {
		// 	// this.props.enableLayers();
		// });
		this.setState({
			map: map
		});
	};

	setPosition = map => {
		const { updatePersistedState } = this.props;
		// convert center object to array
		const newCoords = Object.values(map.getCenter());
		updatePersistedState("cameras-app", "mapSettings", {
			mapCenter: newCoords,
			mapZoom: map.getZoom()
		});
	};

	shouldComponentUpdate(nextProps) {
		if (this.state.map && this.props.WavCamOpen === nextProps.WavCamOpen) {
			return false;
		} else {
			return true;
		}
	}



	getToken() {
		return "pk.eyJ1IjoibGNsYXJrc29uIiwiYSI6ImNpaWh2dzlnaDAwZmp1ZGtueXdpa2loNXcifQ.gcbR_GT1AD2kUmIxkSxseA";
	}

	render() {
		const { zoom, center, WavCamOpen } = this.props;
		const { mapStyle } = this.state;
		return (
			<Map
				ref="map"
				/* Comment out accessToken and style below to use internal tile server */
				style={mapStyle}
				center={center}
				maxZoom={18}
				minZoom={2}
				zoom={[zoom]}
				/* Uncomment this line to use local settings for tile server */
				/* style={simple} */
				containerStyle={{ width: "100vw", height: `calc(100vh - ${WavCamOpen ? "288px" : "48px"})` }}
				onStyleLoad={map => this.setMap(map)}
				className="map-base"
				movingMethod="easeTo"
				onMoveEnd={this.setPosition}
			>
				<ZoomControl />
				<RotationControl />
				{this.state.map && (
					<ErrorBoundary>
						<CamerasMapLayersContainer map={this.state.map} />
						<ContextMenuContainer map={this.state.map} />
					</ErrorBoundary>
				)}
			</Map>
		);
	}
}

export default MapBaseMapboxGL;
