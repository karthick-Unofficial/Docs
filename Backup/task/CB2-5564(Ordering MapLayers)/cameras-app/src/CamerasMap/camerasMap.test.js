import React from "react";
import CamerasMap from "./CamerasMap";
import { shallow } from "enzyme";
import * as reactRedux from 'react-redux';
import { appState, clientConfig } from "orion-components/testData";

jest.mock("react-redux", () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
}));

describe("CamerasMap", () => {
    beforeEach(() => {
        useSelectorMock.mockImplementation(selector => selector(mockStore));
    })
    afterEach(() => {
        useSelectorMock.mockClear();
    })

    const useSelectorMock = reactRedux.useSelector;

    const mockStore = {
        appState,
        baseMaps: {},
        clientConfig,
        mapState: {
            baseMap: {
                mapRef: ""
            }
        }
    };

    it("renders", () => {
        const wrapper = shallow(<CamerasMap />);
        expect(wrapper).toMatchSnapshot();
    });
});