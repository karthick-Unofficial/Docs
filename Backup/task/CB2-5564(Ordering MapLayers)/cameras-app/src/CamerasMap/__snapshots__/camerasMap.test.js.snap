// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CamerasMap renders 1`] = `ShallowWrapper {}`;
